h/api/parse_document_claims.py

```diff
@@ -0,0 +1,237 @@
+        if path_prefix is None:
+            path_prefix = []
+
+        for key, value in items.iteritems():
+            keypath = path_prefix[:]
+            keypath.append(key)
+
+            if isinstance(value, dict):
+                transform_meta_(document_meta_dicts,
+                                value,
+                                path_prefix=keypath)
+            else:
+                if not isinstance(value, list):
+                    value = [value]
+
+                document_meta_dicts.append({
+                    'type': '.'.join(keypath),
+                    'value': value,
+                    'claimant': claimant,
+                })
+
+    items = {k: v for k, v in document_data.iteritems() if k != 'link'}
+    document_meta_dicts = []
+    transform_meta_(document_meta_dicts, items)
+    return document_meta_dicts
+
+
+def document_uris_from_links(link_dicts, claimant):
+    """
+    Return document URI dicts for the given document.link data.
+
+    Process a document.link list of dicts that the client submitted as part of
+    an annotation create or update request and return document URI dicts for
+    all of the document equivalence claims that it makes.
+
+    """
+    document_uris = []
+    for link in link_dicts:
+
+        # Disregard self-claim URLs as they're added separately later.
+        if link.keys() == ['href'] and link['href'] == claimant:
+            continue
+
+        # Disregard doi links as these are being added separately from the
+        # highwire and dc metadata later on.
+        if link.keys() == ['href'] and link['href'].startswith('doi:'):
+            continue
+
+        # Disregard Highwire PDF links as these are being added separately from
+        # the highwire and dc metadata later on.
+        if set(link.keys()) == set(['href', 'type']):
+            if link['type'] == 'application/pdf':
+                continue
+
+        uri_ = link['href']
+        type_ = None
+
+
+        # Handle rel="..." links.
+        if type_ is None and link.get('rel') is not None:
+            type_ = 'rel-{}'.format(link['rel'])
+
+        # The "type" item in link dicts becomes content_type in DocumentURIs.
+        content_type = None
+        if link.get('type'):
+            content_type = link['type']
+
+        document_uris.append({
+            'claimant': claimant,
+            'uri': uri_,
+            'type': type_,
+            'content_type': content_type,
+        })
+
+    return document_uris
+
+
+def document_uris_from_highwire_pdf(highwire_dict, claimant):
+    """
+    Return PDF document URI dicts for the given 'highwire' document metadata.
+
+    Process a document.highwire dict that the client submitted as part of an
+    annotation create or update request and return document URI dicts for all
+    of the PDF document equivalence claims that it makes.
+
+    """
+    document_uris = []
+    hwpdfvalues = highwire_dict.get('pdf', [])
```

Comment:
I think this should be `highwire_dict.get('pdf_url', [])`, as that's what the [annotator does](https://github.com/openannotation/annotator/blob/v1.2.x/src/plugin/document.coffee#L122-L128).
Or did you find some data that had the PDF url with the key `pdf`?